### ISSUE TITLE
Module not found: Can't resolve 'node:os' 

### ISSUE THREAD

**Issue by norgeous** [sindresorhus/fix-path#15]
fix-path: `4.0.0`
electron:  `13.2.3`
@craco/craco: `6.2.0`

Hello, any idea what i'm doing wrong? My app fails to compile after add this package as a dep.

```
[0] Failed to compile.
[0] 
[0] ./node_modules/default-shell/index.js
[0] Module not found: Can't resolve 'node:os' in '[...]/node_modules/default-shell'
```

**Comment by sindresorhus** [comment]
https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c

**Comment by sindresorhus** [comment]
This looks like a problem with your build tool. You should open an issue there instead. `node:os` is a valid module specifier.

**Comment by linonetwo** [comment]
So finally how to solve this? I got `.webpack/main/454.index.worker.js Error: Cannot find module 'node:process'`

**Comment by linonetwo** [comment]
Solution:

based on fix-path 4.0.0

```ts
import { execSync } from 'child_process';
import { userInfo } from 'os';
import process from 'process';
import stripAnsi from 'strip-ansi';

const defaultShell = (() => {
  const { env, platform } = process;
  if (platform === 'win32') {
    return env.COMSPEC ?? 'cmd.exe';
  }
  try {
    const { shell } = userInfo();
    if (shell !== undefined) {
      return shell;
    }
  } catch {}
  if (platform === 'darwin') {
    return env.SHELL ?? '/bin/zsh';
  }
  return env.SHELL ?? '/bin/sh';
})();

const arguments_ = ['-ilc', 'echo -n "_SHELL_ENV_DELIMITER_"; env; echo -n "_SHELL_ENV_DELIMITER_"; exit'];

const environment = {
  // Disables Oh My Zsh auto-update thing that can block the process.
  DISABLE_AUTO_UPDATE: 'true',
};

const parseEnvironment = (environment_: string): Record<string, string> => {
  environment_ = environment_.split('_SHELL_ENV_DELIMITER_')[1];
  const returnValue: Record<string, string> = {};

  for (const line of stripAnsi(environment_)
    .split('\n')
    .filter((line) => Boolean(line))) {
    const [key, ...values] = line.split('=');
    returnValue[key] = values.join('=');
  }

  return returnValue;
};

export function shellEnvironmentSync(shell?: string): NodeJS.ProcessEnv {
  if (process.platform === 'win32') {
    return process.env;
  }

  try {
    const stdout = execSync(`${shell ?? defaultShell} ${arguments_.join(' ')}`, { env: environment });
    return parseEnvironment(String(stdout));
  } catch (error) {
    if (shell !== undefined) {
      throw error;
    } else {
      return process.env;
    }
  }
}

export function shellPathSync(): string | undefined {
  const { PATH } = shellEnvironmentSync();
  return PATH;
}

export function fixPath(): void {
  if (process.platform === 'win32') {
    return;
  }
  process.env.PATH = shellPathSync() ?? ['./node_modules/.bin', '/.nodebrew/current/bin', '/usr/local/bin', process.env.PATH].join(':');
}
```
